lib/msf/core/post/file.rb

```diff
@@ -692,4 +683,43 @@ def _unix_max_line_length
 
     line_max
   end
+
+
+  def stat(filename)
+     if session.type == 'meterpreter'
+      return session.fs.file.stat(filename)
+    end
+    if session.type == 'shell'
+      if session.platform == 'windows'
+        raise "`stat' method does not support Windows systems"
+      end
+      return FileStat.new(filename, session)
+    end
+  end
+
+
+  class FileStat < Rex::Post::FileStat
+  
+    attr_accessor :stathash
+
+    def initialize(filename, session)
+      data = session.shell_command_token("stat --format='%d,%i,%h,%u,%g,%t,%s,%B,%b,%x,%y,%z,%f' #{filename}").to_s.chomp
```

Comment:
Since we're treating timestamps as integers now, this should return them as an epoch. This would prevent you from having to parse the string timestamp.

```suggestion
      data = session.shell_command_token("stat --format='%d,%i,%h,%u,%g,%t,%s,%B,%b,%X,%Y,%Z,%f' #{filename}").to_s.chomp
```

Reply:
:+1:

Reply:
Okay thanks : )

Reply:
Okay :+1: